src/users.cpp

```diff
@@ -758,14 +759,29 @@ bool LocalUser::SetClientIP(const std::string& address, bool recheck_eline)
 
 void LocalUser::SetClientIP(const irc::sockets::sockaddrs& sa, bool recheck_eline)
 {
-	if (sa != client_sa)
-	{
-		User::SetClientIP(sa);
-		if (recheck_eline)
-			this->exempt = (ServerInstance->XLines->MatchesLine("E", this) != NULL);
+	if (sa == client_sa)
+		return;
 
-		FOREACH_MOD(OnSetUserIP, (this));
-	}
+	ServerInstance->Users->RemoveCloneCounts(this);
+
+	User::SetClientIP(sa);
+	if (recheck_eline)
```

Comment:
Is the recheck_eline parameter ever false?

If not, can you move the setting of exempt and the CheckLines call to core_xline?